napari/_qt/qt_main_window.py

```diff
@@ -132,6 +133,20 @@ def __init__(self, viewer: 'Viewer', parent=None) -> None:
                 self._qt_viewer.canvas._backend.screen_changed
             )
 
+        self.status_throttler = QSignalThrottler(parent=self)
+        self.status_throttler.setTimeout(50)
+
+        # Here we disconnect function that update statusbar,
+        # and connect it throttled version to get smother GUI experience
```

Comment:
```suggestion
        # In the GUI we expect lots of changes to the cursor position, so
        # replace the direct connection with a throttled one.
```

The explanation is the key thing here - the code mostly describes itself already.